python/cudf/udf_cpp/groupby/function.cu

```diff
@@ -284,7 +284,7 @@ extern "C" {
   __device__ int name##_##cname(return_type* numba_return_value, type* const data, int64_t size) \
   {                                                                                              \
     return_type const res = name<type>(data, size);                                              \
-    if (threadIdx.x == 0) { *numba_return_value = res; }                                         \
+    *numba_return_value   = res;                                                                 \
     __syncthreads();                                                                             \
```

Comment:
I think the `__syncthreads()` is now redundant, but I am not very confident, since I realise I don't see how these device functions fit together (in terms of how they are called from a numba `__global__` function).

Reply:
Removed

Reply:
And back in...